cmd/gwi-api/api/api_test.go

```diff
@@ -0,0 +1,96 @@
+package api_test
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/tchap/gwi-challenge/cmd/gwi-api/api"
+	"github.com/tchap/gwi-challenge/cmd/gwi-api/api/stores/mockstore"
+
+	"github.com/labstack/echo"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"go.uber.org/zap"
+)
+
+// ----------------------------------------------------------------------------
+// NOTE: Testing a single enpoint for time purposes only.
+// ----------------------------------------------------------------------------
+
+func TestAPI_GetTeamByID(t *testing.T) {
+	assert := assert.New(t)
+
+	// Testing endpoints returning *echo.HTTPError is kinda weird.
+	// You cannot check the response object, you have to check the error object being returned.
+	// So in case ExpectedResponseError is set, we do not check ExpectedResponseStatusCode at all.
+	testCases := []struct {
+		RequestedTeamID            string
+		StoredTeam                 *api.Team
+		ExpectedResponseError      error
+		ExpectedResponseStatusCode int
+		ExpectedResponseBody       echo.Map
+	}{
+		// Team found
+		{
+			RequestedTeamID: "gladiators",
+			StoredTeam: &api.Team{
+				ID:   "gladiators",
+				Name: "The Gladiators",
+			},
+			ExpectedResponseStatusCode: http.StatusOK,
+			ExpectedResponseBody: echo.Map{
+				"id":   "gladiators",
+				"name": "The Gladiators",
+			},
+		},
+		// Team not found
+		{
+			RequestedTeamID:       "gladiators",
+			ExpectedResponseError: echo.ErrNotFound,
+		},
+	}
+
+	for _, tc := range testCases {
```

Comment:
Are you aware of any alternatives to this approach (hint: the `testify` package that you're using offers one). If yes, is there a reason you prefer this one?

Reply:
If you mean organizing tests, you can use suites and you probably mean https://godoc.org/github.com/stretchr/testify/suite . I remember using https://onsi.github.io/ginkgo/ on a project, not sure which one...